src/mumble/MainWindow.cpp

```diff
@@ -506,28 +506,52 @@ bool MainWindow::nativeEvent(const QByteArray &, void *message, long *) {
 void MainWindow::closeEvent(QCloseEvent *e) {
 #ifndef Q_OS_MAC
 	ServerHandlerPtr sh = Global::get().sh;
-	if (sh && sh->isRunning() && Global::get().s.bAskOnQuit && !bSuppressAskOnQuit) {
+	if (sh && sh->isRunning() && !bSuppressAskOnQuit
+		&& Global::get().s.connectedQuitBehaviour == ConnectedQuitBehaviourID::ASK) {
 		QMessageBox mb(QMessageBox::Warning, QLatin1String("Mumble"),
 					   tr("Mumble is currently connected to a server. Do you want to Close or Minimize it?"),
 					   QMessageBox::NoButton, this);
+		QCheckBox   *qcbRemember = new QCheckBox(tr("Remember this setting"));
 		QPushButton *qpbClose    = mb.addButton(tr("Close"), QMessageBox::YesRole);
 		QPushButton *qpbMinimize = mb.addButton(tr("Minimize"), QMessageBox::NoRole);
 		QPushButton *qpbCancel   = mb.addButton(tr("Cancel"), QMessageBox::RejectRole);
 		mb.setDefaultButton(qpbClose);
 		mb.setEscapeButton(qpbCancel);
+		mb.setCheckBox(qcbRemember);
 		mb.exec();
 		if (mb.clickedButton() == qpbMinimize) {
 			showMinimized();
 			e->ignore();
+
+			if (qcbRemember->isChecked()) {
+				Global::get().s.connectedQuitBehaviour = ConnectedQuitBehaviourID::MINIMIZE;
+			}
+
 			return;
 		} else if (mb.clickedButton() == qpbCancel) {
 			e->ignore();
 			return;
 		}
+
+		if (qcbRemember->isChecked()) {
+			Global::get().s.connectedQuitBehaviour = ConnectedQuitBehaviourID::CLOSE;
+		}
+	} else if (Global::get().s.connectedQuitBehaviour == ConnectedQuitBehaviourID::MINIMIZE) {
```

Comment:
Now this does something else: This not only minimizes the application, when the user is still connected to a server, but every time, the user clicks on the X - regardless of whether a server connection exists or not.
However, the setting this PR is about is - to my understanding - specifically about the behavior when quitting whilst connected to a server. In that context, this implementation here would be wrong.

I think an always-minimize-when-closing might be a good addition in its own right (if something like this doesn't exist yet), but that would (imo) be a separate feature.

I would suggest moving the checks for the different possible values of `connectedQuitBehavior` into a `switch` statement inside the above if-block that checks whether the user is still connected to a server.